code/UMLModel.py

```diff
@@ -337,17 +313,18 @@ def load_model(self, filename):
             NOTE: File should be a JSON file generated by save_model() to
                 ensure parsing is correct 
         """
-        # Clear out previous model
-        self.classes = {}
-
-        # Holds the data loaded from json
-        raw_model = {}
 
         # Ensure file exists
         if not path.exists(MODEL_DIRECTORY+filename):
             print (f"{filename} does not exist")
             return
 
+        # Clear out previous model
+        self.classes = {}
+
+        # Holds the data loaded from json
+        raw_model = {}
+
```

Comment:
I moved these lines so that the model is not cleared if the filename is rejected